src/components/VerifiedFieldComponents.tsx

```diff
@@ -0,0 +1,174 @@
+import CheckIcon from "@mui/icons-material/Check";
+import {
+  Box,
+  Checkbox,
+  Divider,
+  IconButton,
+  InputBase,
+  Tooltip,
+  Typography,
+} from "@mui/material";
+import { ReactNode, useState } from "react";
+import { Control, Controller, useFormContext, useWatch } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+
+function VerifiedFieldWrapper({
+  label,
+  path,
+  className = "",
+  renderField,
+}: {
+  label: string;
+  path: string;
+  className?: string;
+  renderField: (props: {
+    setIsFocused: (value: boolean) => void;
+    control: Control;
+    valuePath: string;
+    verified: boolean;
+  }) => ReactNode;
+}) {
+  const { t } = useTranslation("labelDataValidationPage");
+  const { control } = useFormContext();
+  const [isFocused, setIsFocused] = useState(false);
+
+  const valuePath = `${path}.value`;
+  const verifiedPath = `${path}.verified`;
+  const verified: boolean = useWatch({
+    control,
+    name: verifiedPath,
+  });
+
+  return (
+    <Box
+      className={`flex items-center p-1 border-2 rounded-tr-md rounded-br-md ${
+        isFocused ? "border-fertiscan-blue" : ""
+      } ${className}`}
+      data-testid={`verified-field-${path}`}
+    >
+      <Typography
+        className="px-2 !font-bold select-none"
+        data-testid={`field-label-${path}`}
+      >
+        {label}
+      </Typography>
+      {renderField({ setIsFocused, control, valuePath, verified })}
+      <Divider
+        orientation="vertical"
+        flexItem
+        className={isFocused ? "!border-fertiscan-blue" : ""}
+        data-testid={`divider-${path}`}
+      />
+      <Controller
+        name={verifiedPath}
+        control={control}
+        render={({ field: { value, onChange } }) => (
+          <Tooltip
+            title={
+              verified
+                ? t("verifiedField.unverify", { label })
+                : t("verifiedField.verify", { label })
+            }
+            enterDelay={1000}
+          >
+            <IconButton
+              onClick={() => onChange(!value)}
+              data-testid={`toggle-verified-btn-${verifiedPath}`}
+              aria-label={
+                verified
+                  ? t("verifiedField.unverify", { label })
+                  : t("verifiedField.verify", { label })
+              }
+            >
+              <CheckIcon
+                className={value ? "text-green-500" : ""}
+                data-testid={`verified-icon-${verifiedPath}`}
+                aria-hidden
+              />
+            </IconButton>
+          </Tooltip>
+        )}
+      />
+    </Box>
+  );
+}
+
+function VerifiedCheckbox({
+  label,
+  path,
+  className = "",
+}: {
+  label: string;
+  path: string;
+  className?: string;
+}) {
+  const { t } = useTranslation("labelDataValidationPage");
```

Comment:
Use arrow function syntax

Define an interface instead of having it inline for readability / consistency with other functions

Reply:
done